heron/uploaders/src/java/com/twitter/heron/uploader/s3/sample.yaml

```diff
@@ -9,13 +9,24 @@ heron.uploader.s3.bucket:     heron-topologies-company-com
 # s3://#{bucket}/#{prefix}/#{topology_name}/topology.tar.gz
 # heron.uploader.s3.path_prefix: path/prefix
 
-# By default, assume AWS S3. However, you can specify custom url if you are using proxy or a
-# S3 compatible storage layer.
-# heron.uploader.s3.uri : localhost:9000
+# By default, assume AWS S3. However, you can specify a custom url if you are using a
+# S3 compatible storage layer (or using a reverse proxy for accessing S3).
+# heron.uploader.s3.uri: hostname:port
+# If you want to access S3 through a http proxy, use
+# heron.upload.s3.proxy_uri: http://username:password@hostname:port
+# username and password are optional
+
 # Specifies a custom region - see http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/regions/Regions.html#US_EAST_1
 # heron.uploader.s3.region : meh
 
-# Credentials that have write access to the bucket specified above
-heron.uploader.s3.access_key: access_key
-heron.uploader.s3.secret_key: secret_access_key
+# AWS Credentials
+# By default the S3 Uploader will use the Default Credential Provider Chain for accessing the S3 bucket -
+# see http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#using-the-default-credential-provider-chain
+# However you have the option to directly specify aws credentials:
+# heron.uploader.s3.access_key: access_key
+# heron.uploader.s3.secret_key: secret_access_key
+# If you have multiple AWS profiles in your credentials file (~/.aws/credentials)
```

Comment:
We should clarify that if both access_key, secret_key and aws_profile are all included the latter will be ignored. Or probably better to throw an exception if all 3 are specified, just to make sure things are clear. That's consistent with how we throw an exception if just aws_profile and access_key are set for example.